### ISSUE TITLE
Heart of Avelorn for Tyrion still not being granted still

### ISSUE THREAD

**Issue by missj4** [chadvandy/cbfm_wh3#572]
**Describe the bug**
Was stated as likely fixed/closed, played Tyrion, did not get granted Heart of Avelorn as intended at the right level. Was allied to Avelorn, they got the gate and did not receive the item. mission was aborted, multiple people on steam bug report stating same bug issue.

**Expected behavior**
if the region is owned by an ally it should grant the item, but it does not.

**To Reproduce**
1. Play as Tyrion
2. get to level to unlock quest for heart of avelorn
3. Quest is disbanded immediately upon reaching required level and not owning Phoenix Gate (guessing if you are not allied, the quest might be granted then you could ally and try it but the time it takes to get to lv17 I havent tried this route)

**Comment by missj4** [comment]
Yes, version 2.21, started November 8, 2022. My Tyrion is lv 23, but I don't have the item, i don't even recall a popup for it to act upon. I could probably find in one of the auto saves and test my theory that way, annex them and see if I get the item on level up or double check if you need.

**Comment by Harmandodelavega** [comment]
we good

**Comment by Thurhame** [comment]
Works fine for me. Game version 2.3, I leveled up and got the quest after Avelorn already held Phoenix Gate. I forget if it was that turn, next turn or a few turns later, but quest was automatically cancelled and I got the Heart. Now I just need Sunfang :(
